File: app/src/main/java/com/android/ichooseyou/activity/OptionsAdapter.java
package com.android.ichooseyou.activity;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.android.ichooseyou2.R;  // Add this import
import java.util.List;

public class OptionsAdapter extends RecyclerView.Adapter<OptionsAdapter.OptionViewHolder> {

    private List<String> options;

    public OptionsAdapter(List<String> options) {
        this.options = options;
    }

    @NonNull
    @Override
    public OptionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_option, parent, false);
        return new OptionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull OptionViewHolder holder, int position) {
        holder.optionText.setText(options.get(position));
    }

    @Override
    public int getItemCount() {
        return options.size();
    }

    static class OptionViewHolder extends RecyclerView.ViewHolder {
        TextView optionText;

        OptionViewHolder(View itemView) {
            super(itemView);
            optionText = itemView.findViewById(R.id.option_text);
        }
    }
}

File: app/src/main/java/com/android/ichooseyou/activity/OptionsAdapter.kt
package com.android.ichooseyou.activity

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.android.ichooseyou.activity.OptionsAdapter.OptionViewHolder
import com.android.ichooseyou2.R
class OptionsAdapter(private val options: List<String>) : RecyclerView.Adapter<OptionViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OptionViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_option, parent, false)
        return OptionViewHolder(view)
    }

    override fun onBindViewHolder(holder: OptionViewHolder, position: Int) {
        holder.optionText.text = options[position]
    }

    override fun getItemCount(): Int {
        return options.size
    }

    class OptionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var optionText: TextView

        init {
            optionText = itemView.findViewById(R.id.option_text)
        }
    }
}